sources/models/src/lib.rs

```diff
@@ -90,3 +91,55 @@ struct Report {
     name: String,
     description: String,
 }
+
+/// Strength represents whether the settings can be overridden by a setting generator or not.
+#[derive(Deserialize, Serialize, Debug, Clone, Copy, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+pub enum Strength {
+    Strong,
+    Weak,
+}
+
+impl std::fmt::Display for Strength {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Strength::Strong => write!(f, "strong"),
+            Strength::Weak => write!(f, "weak"),
+        }
+    }
+}
+
+derive_fromstr_from_deserialize!(Strength);
+
+/// Struct to hold the setting generator definition containing
+/// command, strength, skip-if-populated
+#[derive(Deserialize, Serialize, std::fmt::Debug, PartialEq)]
+pub struct SettingsGenerator {
+    pub command: String,
+    pub strength: Strength,
+    #[serde(rename = "skip-if-populated")]
+    pub skip_if_populated: bool,
+}
```

Comment:
I have a few suggestions:

```suggestion
/// Struct to hold the setting generator definition containing
/// command, strength, skip-if-populated
#[derive(Deserialize, Serialize, std::fmt::Debug, PartialEq)]
#[serde(rename_all = "kebab-case", deny_unknown_fields)]
pub struct SettingsGenerator {
    pub command: String,
    #[serde(default = Strength::Strong)]
    pub strength: Strength,
    #[serde(default = true)]
    pub skip_if_populated: bool,
}
```

This would allow you to specify only the command if desired, falling back to the defaults that exist for the current generator semantics. I think `deny_unknown_fields` can also be useful for guarding against typos that are silently accepted.